lib/order.rb

```diff
@@ -1,18 +1,148 @@
+require 'csv'
+
 module Grocery
   class Order
     attr_reader :id, :products
 
+    #@@allordersnumber = 0
+    #@@allorders = []
+
     def initialize(id, products)
       @id = id
       @products = products
+      #@allorders = []
+      #@@allordersnumber += 1
+      #@@allorders << self
     end
 
+
     def total
-      # TODO: implement total
+      without_taxes = 0
+      @products.each do |product, value|
+        without_taxes += value
+      end
+      total_price = without_taxes + ( without_taxes * 0.075)
+      return total_price.round(2)
     end
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      if @products.key?(product_name)
+        return false
+      else
+        @products[product_name] = product_price
+        return true
+      end
+    end
+
+    def delete_product(product_name)
+      if @products.key?(product_name)
+        @products.delete(product_name)
+        return true
+      else
+        return false
+      end
+    end
+
+    def see
+      return @products
+    end
+
+    def self.all
+      csv_order = {}
+      allorders = []
+
+      CSV.open('support/orders.csv', "r").each do |row|
+        csv_order[row[0].to_i] = row[1]
+      end
+
+
+      temp={}
+      csv_order.each do |k ,v|
+        temp[k] = v.split(";")
```

Comment:
Good work figuring out the logic to parse the product list.

This code ends up being pretty complicated, which makes it essential to have descriptive variable names. If you or a coworker comes through in the future to try and add a feature or fix a bug, this will help them understand how the pieces fit together.

`temp` or `temp1` doesn't tell the reader anything about what kind of variable this is or what it stores. What about something like `product_arrays` for `temp` or `product_hashes` for `temp1`? Even better, you could leave a comment detailing what you're doing in each section and why.